### ISSUE TITLE
Twitch VoD (past streams) unhandled exception errors, randomly

### ISSUE THREAD

**Issue by Fironet** [yt-dlp/yt-dlp#6337]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2023.02.17** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Provide a description that is worded well enough to be understood

Twitch video on demand (VoD) videos of past broadcasts have been failing with unhandled exceptions, randomly, when either checking the available streams or attempting to download them, in the past number of days.

As an example of the inconsistency, when creating this ticket I repeatedly reproduced the unhandled errors, then when adding the `-vU` verbose flag it suddenly worked again *once*, before subsequently erroring when repeating the same command.

For the Complete Verbose Output the instance where the unhandled error was encountered was copied.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', '-F', 'https://www.twitch.tv/videos/1747210136']
[debug] Portable config "D:\Global\yt-dlp.conf": ['-o', 'M:\\Videos\\%(title)s [%(uploader)s, %(extractor)s].%(ext)s', '--console-
title', '--no-mtime', '--add-metadata', '--compat-options', 'all', '--no-clean-infojson', '--compat-option', 'filename-sanitizatio
n']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8 (No VT), error utf-8 (No VT), screen utf-8 (No VT)
[debug] yt-dlp version 2023.02.17 [a0a7c01] (win_exe)
[debug] Compatibility options: playlist-index, no-direct-merge, no-attach-info-json, *no-clean-infojson, embed-thumbnail-atomicpar
sley, multistreams, seperate-video-versions, no-youtube-unavailable-videos, no-live-chat, format-sort, filename-sanitization, no-c
ertifi, format-spec, no-playlist-metafiles, no-youtube-channel-redirect, list-formats, no-external-downloader-progress, embed-meta
data, abort-on-error, no-youtube-prefer-utc-upload-date, *filename, no-keep-subs
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-8.1-6.3.9600-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg N-105349-g0a83ecbf48-20220124 (setts), ffprobe N-105349-g0a83ecbf48-20220124, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1782 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2023.02.17, Current version: 2023.02.17
yt-dlp is up to date (2023.02.17)
[twitch:vod] Extracting URL: https://www.twitch.tv/videos/1747210136
[twitch:vod] 1747210136: Downloading stream metadata GraphQL
Traceback (most recent call last):
  File "yt_dlp\__main__.py", line 17, in <module>
  File "yt_dlp\__init__.py", line 967, in main
  File "yt_dlp\__init__.py", line 957, in _real_main
  File "yt_dlp\YoutubeDL.py", line 3353, in download
  File "yt_dlp\YoutubeDL.py", line 3328, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1491, in extract_info
  File "yt_dlp\YoutubeDL.py", line 1502, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1578, in __extract_info
  File "yt_dlp\extractor\common.py", line 693, in extract
  File "yt_dlp\extractor\twitch.py", line 516, in _real_extract
  File "yt_dlp\extractor\twitch.py", line 384, in _download_info
TypeError: 'NoneType' object does not support item assignment
[4744] Failed to execute script '__main__' due to unhandled exception!
```

**Comment by Tharn** [comment]
Getting the same thing with using MPC-HC and yt-dlp to watch VODs.
`ERROR: "NoneType" object does not support item assignment.`

**Comment by bashonly** [comment]
Duplicate of #6308

Will be fixed by #6318
